inferred-spans/src/main/java/co/elastic/otel/profiler/asyncprofiler/JfrParser.java

```diff
@@ -105,169 +105,195 @@ public void parse(
     this.includedClasses = includedClasses;
     bufferedFile.setFile(file);
     long fileSize = bufferedFile.size();
-    if (fileSize < 16) {
+
+    int chunkSize = readChunk(0);
+    if (chunkSize < fileSize) {
       throw new IllegalStateException(
-          "Unexpected sampling profiler error, everything else should work as expected. "
-              + "Please report to us with as many details, including OS and JVM details.");
+          "This implementation does not support reading JFR files containing multiple chunks");
```

Comment:
[question] does it means we could make it work (probably more efficiently) with >1 chunk per file in the future ?